src/learning/modules.py

```diff
@@ -0,0 +1,113 @@
+import torch.nn as nn
+import utils
+
```

Comment:
Not very important but maybe we could rename this file to something like nn_modules.py? Might be more explicit since there are also Python modules and the new name would make it clear what the file contains.

Reply:
Done.